pytest_plugins/rerun_rp/rerun_rp.py

```diff
@@ -14,40 +14,69 @@ class LaunchError(Exception):
     pass
 
 
-def _get_failed_tests(launch, fail_args):
-    """Returns failed test names from Report Portal Launch based on arguments"""
-    test_args = (
-        [dict(status='failed')]
-        if fail_args == ['all']
-        else [dict(defect_type=dtype) for dtype in fail_args]
-    )
-    tests = []
-    for t_args in test_args:
-        tests.extend([*launch.tests(**t_args).keys()])
-    # Formating test names for 'member of pytest test items' operation
+def _transform_rp_tests_to_pytest(tests):
+    """Formating test names toc check 'member of pytest test items' operation
+
+    :param list tests: The list of tests those name to be changed
+    :return list: The transformed list of tests
+    """
     tests = [str(test).replace('::', '.') for test in tests]
     return tests
 
 
-def _get_test_collection(rp_failed_tests, items):
+def _get_tests(launch, **kwargs):
+    """The helper function that actually connects to Report Portal and fetch tests of a launch
+
+    :param rp.launch launch: The launch object from where the tests will be read
+    :param dict kwargs: The params to `launch.tests` function
+        i.e status, user, defect_types
+    :return list: The list of test names retrieved from RP launch
+    """
+    tests = []
+    if 'defect_types' in kwargs:
+        defect_types = kwargs.pop('defect_types')
+        for dtype in defect_types:
+            tests.extend([*launch.tests(**kwargs, defect_type=dtype).keys()])
+    else:
+        tests.extend([*launch.tests(**kwargs).keys()])
```

Comment:
I believe this should work
```suggestion
            tests.extend(launch.tests(**kwargs, defect_type=dtype).keys())
    else:
        tests.extend(launch.tests(**kwargs).keys())
```

Reply:
It works ! :)